tests/impersonation.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# from __future__ import print_function
+import errno
+import os
+import subprocess
+import unittest
+
+from airflow import jobs, models
+from airflow.utils.state import State
+from datetime import datetime
+
+DEV_NULL = '/dev/null'
+TEST_DAG_FOLDER = os.path.join(
+    os.path.dirname(os.path.realpath(__file__)), 'dags')
+DEFAULT_DATE = datetime(2015, 1, 1)
+TEST_USER = 'airflow_test_user'
+
+
+# TODO(aoen): Adding/remove a user as part of a test is very bad (especially if the user
+# already existed to begin with on the OS), this logic should be moved into a test
+# that is wrapped in a container like docker so that the user can be safely added/removed.
+# When this is done we can also modify the sudoers file to ensure that useradd will work
+# without any manual modification of the sudoers file by the agent that is running these
+# tests.
+
+class ImpersonationTest(unittest.TestCase):
```

Comment:
I think it's acceptable until we have containers. Usually users are running specific tests on dev systems anyways (since the full gauntlet fails), so it would only be run when users want to explicitly test this. Even if they did run the full suite it would probably fail on lack of sudo perms (or maybe it would succeed but that's ok too).